pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -303,15 +302,48 @@ async def main():
     If `None`, no log level will be set.
     """
 
+    def __post_init__(self):
+        if not isinstance(self.timeout, timedelta):
+            warnings.warn(
+                'Passing timeout as a float has been deprecated, please use a timedelta instead.',
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.timeout = timedelta(seconds=self.timeout)
+
+        if not isinstance(self.sse_read_timeout, timedelta):
+            warnings.warn(
+                'Passing sse_read_timeout as a float has been deprecated, please use a timedelta instead.',
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.sse_read_timeout = timedelta(seconds=self.sse_read_timeout)
+
     @asynccontextmanager
     async def client_streams(
         self,
     ) -> AsyncIterator[
-        tuple[MemoryObjectReceiveStream[JSONRPCMessage | Exception], MemoryObjectSendStream[JSONRPCMessage]]
+        tuple[MemoryObjectReceiveStream[SessionMessage | Exception], MemoryObjectSendStream[SessionMessage]]
     ]:  # pragma: no cover
-        async with sse_client(
+        if not isinstance(self.timeout, timedelta):
```

Comment:
I duplicated these checks both to make the typechecking happy and to avoid a breaking change if someone was modifying the server after instantiation.